### PR TITLE
feat(AvroConsumer): create new class

### PR DESCRIPTION
**Pull request by omamiya**

### CODE REVIEW

File: lib/yotpo_kafka/avro_consumer.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'avro_turf/messaging'
+
+module YotpoKafka
+  class AvroConsumer < YotpoKafka::BaseConsumer
+    extend Ylogger
+
+    attr_accessor :kafka
+
+    def initialize(params = {})
+      @avro_messaging = AvroTurf::Messaging.new(registry_url: ENV['REGISTRY_URL'] || 'schema-registry.us-east-1.yotpo.xyz')
```
Comment (emaymon) [inline lib/yotpo_kafka/avro_consumer.rb]: move to DEFAULT_REGISTRY constant

File: lib/yotpo_kafka/base_consumer.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
+require 'kafka'
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: taif *2

File: lib/yotpo_kafka/avro_consumer.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+require 'avro_turf/messaging'
+
+module YotpoKafka
+  class AvroConsumer < YotpoKafka::BaseConsumer
+    extend Ylogger
```
Comment (emaymon) [inline lib/yotpo_kafka/avro_consumer.rb]: I think you can remove and the attr

File: lib/yotpo_kafka/base_consumer.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    def initialize(params = {})
+      @use_logstash_logger = params[:logstash_logger] != false
+      YotpoKafka::YLoggerKafka.config(@use_logstash_logger)
+      set_log_tag(:yotpo_ruby_kafka)
+      @seed_brokers = params[:broker_url] || ENV['BROKER_URL'] || '127.0.0.1:9092'
+      @kafka = Kafka.new(@seed_brokers)
+      @seconds_between_retries = params[:seconds_between_retries] || 0
+
+      @json_parse = params[:json_parse].nil? ? true : params[:json_parse]
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: extract

File: lib/yotpo_kafka/base_consumer.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      if !YotpoKafka.kafka_v2 && !@json_parse
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: extract to set_retry_policy

File: lib/yotpo_kafka/base_consumer.rb [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                broker_url: @seed_brokers)
+    end
+
+    def subscribe_to_topics
+      @topics.each do |topic|
+        @consumer.subscribe(topic)
+        log_info('Consumer subscribes to topic: ' + topic)
+        next unless @listen_to_failures
+
+        subscribe_to_failure_topic(topic)
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: subscribe_to_failure_topic(topic) unless @listen_to_failures

File: lib/yotpo_kafka/base_consumer.rb [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+      raise 'Could not initialize'
+    end
+
+    def start_consumer
+      log_debug('Starting consume', broker_url: @seed_brokers)
+      subscribe_to_topics
+      @consumer.each_message do |message|
+        @consumer.mark_message_as_processed(message)
+        @consumer.commit_offsets
+        # remember that we got something with key and ignore reties for same key
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: remove

File: lib/yotpo_kafka/base_consumer.rb [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+      YotpoKafka::YLoggerKafka.config(@use_logstash_logger)
+      set_log_tag(:yotpo_ruby_kafka)
+      @seed_brokers = params[:broker_url] || ENV['BROKER_URL'] || '127.0.0.1:9092'
+      @kafka = Kafka.new(@seed_brokers)
+      @seconds_between_retries = params[:seconds_between_retries] || 0
+
+      @json_parse = params[:json_parse].nil? ? true : params[:json_parse]
+      if !YotpoKafka.kafka_v2 && !@json_parse
+        @listen_to_failures = false
+        log_info('retry mechanism not supports messages without header and without json parsing, setting listen_to_failures=false')
```
Comment (emaymon) [inline lib/yotpo_kafka/base_consumer.rb]: bad log

File: lib/yotpo_kafka/base_consumer.rb (hunk 6 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: subscribe_to_failure_topic(topic) *if* @listen_to_failures

File: lib/yotpo_kafka/avro_consumer.rb (hunk 1 above)
Comment (omamiya) [reply lib/yotpo_kafka/avro_consumer.rb]: done

File: lib/yotpo_kafka/base_consumer.rb (hunk 2 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: done

File: lib/yotpo_kafka/avro_consumer.rb (hunk 3 above)
Comment (omamiya) [reply lib/yotpo_kafka/avro_consumer.rb]: done

File: lib/yotpo_kafka/base_consumer.rb (hunk 5 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: done

File: lib/yotpo_kafka/base_consumer.rb (hunk 6 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: done

File: lib/yotpo_kafka/base_consumer.rb (hunk 7 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: done

File: lib/yotpo_kafka/base_consumer.rb (hunk 8 above)
Comment (omamiya) [reply lib/yotpo_kafka/base_consumer.rb]: done

**Pull request merged**